pkg/compose/create.go

```diff
@@ -139,71 +163,139 @@ func (s *composeService) ensureNetworks(ctx context.Context, networks types.Netw
+					continue
+				}
+				p := filepath.Clean(vol.Source)
+				if !filepath.IsAbs(p) {
+					return fmt.Errorf("file share path is not absolute: %s", p)
+				}
+				if fi, err := os.Stat(p); errors.Is(err, fs.ErrNotExist) {
+					// actual directory will be implicitly created when the
+					// file share is initialized if it doesn't exist, so
+					// need to filter out any that should not be auto-created
+					if vol.Bind != nil && !vol.Bind.CreateHostPath {
+						logrus.Debugf("Skipping creating file share for %q: does not exist and `create_host_path` is false", p)
 						continue
 					}
-
-					paths = append(paths, p)
+				} else if err != nil {
+					// if we can't read the path, we won't be able to make
+					// a file share for it
+					logrus.Debugf("Skipping creating file share for %q: %v", p, err)
+					continue
+				} else if !fi.IsDir() {
+					// ignore files & special types (e.g. Unix sockets)
+					logrus.Debugf("Skipping creating file share for %q: not a directory", p)
+					continue
 				}
+
+				paths = append(paths, p)
 			}
+		}
 
-			// remove duplicate/unnecessary child paths and sort them for predictability
-			paths = pathutil.EncompassingPaths(paths)
-			sort.Strings(paths)
+		// remove duplicate/unnecessary child paths and sort them for predictability
+		paths = pathutil.EncompassingPaths(paths)
+		sort.Strings(paths)
 
-			fileShareManager := desktop.NewFileShareManager(s.desktopCli, project.Name, paths)
-			if err := fileShareManager.EnsureExists(ctx); err != nil {
-				return fmt.Errorf("initializing file shares: %w", err)
+		fileShareManager := desktop.NewFileShareManager(s.desktopCli, project.Name, paths)
+		if err := fileShareManager.EnsureExists(ctx); err != nil {
+			return fmt.Errorf("initializing file shares: %w", err)
+		}
+	}
+	return nil
+}
+
+func (s *composeService) recreateVolume(ctx context.Context, project *types.Project, volumeID string, volumeConfig types.VolumeConfig) error {
+	// need to recreate containers associated with the volume first
+	selectedService := []string{}
+	for _, s := range project.Services {
+		for _, v := range s.Volumes {
+			if v.Source == volumeID {
+				selectedService = append(selectedService, s.Name)
 			}
 		}
-		return nil
-	}()
+	}
+	err := s.Remove(ctx, project.Name, api.RemoveOptions{
+		Project:  project,
+		Services: selectedService,
+		Volumes:  true, // this does not work in the apiClient (?)
+		Force:    true,
+		Stop:     true,
+	})
+	if err != nil {
+		return err
+	}
+	// remove volume to update it
+	w := progress.ContextWriter(ctx)
+	err = s.removeVolume(ctx, volumeConfig.Name, w)
+	if err != nil {
+		return err
+	}
+
+	// create the volume
+	return s.createVolume(ctx, volumeConfig)
+}
 
+func shouldUpdateVolume(config types.VolumeConfig, volume *volumetypes.Volume) (bool, error) {
+	previousHash, ok := volume.Labels[api.VolumeConfigHashLabel]
+	if !ok {
+		return false, fmt.Errorf("could not get hash label for volume %s", volume.Name)
```

Comment:
this should not return an error but just ignore, as user who bump compose version will get networks without labels, and will suddenly get all application recreated without a notice -> prepare for a bunch of angry users :P